containers/src/Data/Map/Internal.hs

```diff
@@ -2111,6 +2111,16 @@ disjoint (Bin _ k _ l r) t
 -- 'compose' that forced the values of the output 'Map'. This version does not
 -- force these values.
 --
+-- __Note:__ This is asymptotically optimal.
```

Comment:
Could we add this add a collapsible header? It is fine to have the information but the average user looking at the docs will likely not care about this and shouldn't have to read this.